src/repos/UserRepo.ts

```diff
@@ -58,8 +60,13 @@ const getUserByUUID = async (uuid: string): Promise<User> => {
 /**
  * Return all users who follow a publication.
  */
-const getUsersFollowingPublication = async (pubID: PublicationID): Promise<User[]> => {
-  const users = await UserModel.find({ followedPublications: pubID });
+const getUsersFollowingPublication = async (pubSlug: string): Promise<User[]> => {
+  const publication = await PublicationRepo.getPublicationBySlug(pubSlug);
+  // WARNING TODO: linear scan on DB, inefficient <-- turn into a query that just gets users following publications
+  const users = await UserModel.find();
+  users.filter((u) => {
+    u.followedPublications.map((id) => id.id).includes(publication.id);
+  });
   return users;
 };
```

Comment:
To make this more efficient, can you create a field on the publication for which users follow that publication? I know it's a space tradeoff, but right now your scan is even worse than linear because with N users and they each follow M publications, this takes O(N*M) time. With the field on the publication of the users that follow, you can just return that field which would be constant time

Reply:
M is actually constant(only 15 publications) so its O(N) - i took 2110 too buddy ;), however, I agree with you its inefficient. Particularly because of the query on the database for all users. Will be fixing this.